Add style_guides dir, extracted sections from STYLEGUIDE.md.

cjcenizal opened the pull request:
Migrating from https://gist.github.com/cjcenizal/10a7651732433159c1ba18e4feedce7c

w33ble commented:
I think it makes sense to put this into the existing [STYLEGUIDE.md](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md) file.

cjcenizal commented:
@w33ble I didn't even notice that!

bevacqua commented:
@w33ble Maybe just a link, it makes not a lot of sense to have all the styleguides crammed up together

cjcenizal commented:
I'm inclined to agree with @bevacqua, so I extracted the JS, HTML, and API portions from STYLEGUIDE.md and moved them to `styleGuides/`. STYLEGUIDE.md now links to those pages and contains notes on styles specific to the Kibana project organization. @w33ble thoughts?

epixa commented:
With the exception of `installedPlugins` (for BC reasons), we use underscores for directory and file names, so let's keep this consistent here as well.

epixa commented on styleGuides/apiStyleGuide.md:
```diff
@@ -0,0 +1,32 @@
+API routes must start with the `/api/` path segment, and should be followed by the plugin id if applicable:
+
+*Right:* `/api/marvel/v1/nodes`
+*Wrong:* `/marvel/api/v1/nodes`
```
Let's ditch the version in our example since the very next section discourages versioning.

bevacqua commented on style_guides/api_style_guide.md:
```diff
@@ -0,0 +1,32 @@
+
+# Api Style Guide
```
API

epixa commented on styleGuides/htmlStyleGuide.md:
```diff
@@ -0,0 +1,18 @@
+
+## Multiple attribute values
+
+When a node has multiple attributes that would cause it to exceed the line character limit, each attribute including the first should be on its own line with a single indent. Also, when a node that is styled in this way has child nodes, there should be a blank line between the opening parent tag and the first child tag.
```
This doesn't mention what the line character limit is?

w33ble commented:
I'm assuming that the CSS styleguide is more of a living document than a static set of rules right now. 

In any event, I like how all the guides are broken up. LGTM!
